core/src/main/scala/org/apache/spark/storage/BlockObjectWriter.scala

```diff
@@ -123,12 +133,30 @@ private[spark] class DiskBlockObjectWriter(
    */
   private var numRecordsWritten = 0
 
+  private var sparkConf:SparkConf = null
+
   override def open(): BlockObjectWriter = {
     if (hasBeenClosed) {
       throw new IllegalStateException("Writer already closed. Cannot be reopened.")
     }
     fos = new FileOutputStream(file, true)
-    ts = new TimeTrackingOutputStream(fos)
+    val isEncryptedShuffle = sparkConf.getBoolean("spark.encrypted.shuffle", false)
```

Comment:
instead of implicitly suggesting that the creator of the object of this class must call setSparkConf (otherwise, it will throw a NPE), maybe we shall allow sparkConf be null here?